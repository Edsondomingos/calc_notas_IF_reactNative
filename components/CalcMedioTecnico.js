import * as React from 'react';
import { useState } from 'react';
import { Text, View, StyleSheet, TextInput, Button, TouchableHighlight, ScrollView } from 'react-native';
import Styles from '../assets/styles'

let finall
let media
let media1
let media2
let mediaFinal
let mdf1
let mdf2
let mdf3

export default function CalcSuperior(){
  const [nota1, setNota1] = useState('')
  const [nota2, setNota2] = useState('')
  const [nota3, setNota3] = useState('')
  const [nota4, setNota4] = useState('')
  const [final, setFinal] = useState('')
  const [resultado, setResultado] = useState()
  
  const Calcular = () => {
    let n1 = nota1
    let n2 = nota2
    let n3 = nota3
    let n4 = nota4
    media1 = ((n1 * 2) + (n2 * 2)) 
    // media1 = (240 - (nota1*2))/2 
    // Se estiver sem nota1
    if(nota1.length == 0){
      setResultado('Digite pelo menos a primeira nota')
    } //else {    
      // Se estiver apenas coma primeira nota
      if(nota1.length != 0 && nota2.length == 0 &&
        nota3.length == 0 && nota4.length == 0
      ){        
        setNota2((240 - (nota1*2))/2)
        // setNota3(400/6*3)
        setNota3(60)
        // setNota3(((media1)*3)) 
        // 1000/10*3-media1
        // ((n1*2 + n2*2)/10)*3
        // setNota3((((400 - media1)/60)*3))
        setNota4(((360 - (nota3*3))/3)/2)
        // if(nota3.length == 0 && nota4.length == 0){

        // }
      } else if(nota1.length != 0 && nota2.length != 0 &&
        nota3.length == 0 && nota4.length == 0){
          setNota3(((media1*4) - 360)/10)
          setNota4(((360 - (nota3*3))/3)/2)
      } else if(nota1.length != 0 && nota2.length != 0 &&
        nota3.length != 0 && nota4.length == 0){
          setNota4(((360 - (nota3*3))/3)/2)
          // setNota4((((400- (media1+nota3))*3)/10)*2)
      }

       if (nota1.length != 0 && nota2.length != 0 &&
          nota3.length != 0 && nota4.length != 0){
          
          media2 = ((nota3 * 3) + (nota4 *3)) 
          mediaFinal = (media1 + media2) / 10
          // mediaFinal = ((n1 * 2) + (n2 * 2) + (nota3 * 3) + (nota4 *3)) /10
        if(mediaFinal > 20 && mediaFinal < 60){
          if(media1 < 60){
            // finall = (((300 - (n1*2))/3))
            // setFinal(finall)
            media2
            mdf2 = ('NAF'*2 + (media2 * 3)) / 10
            setResultado('Aprovado com media: '+ mdf2)
          } else if(media2 < 60){
            mdf3 = ((2 * media1) + (3 * 'NAF')) / 10
            setResultado('Aprovado com media: '+ mdf3)
          } else {
            mdf1 = (mediaFinal + 'NAF') / 2
            setResultado('Aprovado com media: '+ mdf1)
          }
          setFinal(mediaFinal)
          setResultado()
        } else if(mediaFinal < 20){
          setResultado('Reprovado por media: '+ mediaFinal)
          setFinal()
        } else {
          setResultado('Aprovado com media: '+ mediaFinal)
          setFinal()
        }      
      }
      // setResultado((((media2*3)-720)/2))
    //}
  }

  function Limpar(){
    setNota1('')
    setNota2('')
    setNota3('')
    setNota4('')
    setFinal('')
    setResultado('')
  }

  return (
    <View style={Styles.container} testID='btn1'>
      <ScrollView>
        <Text style={Styles.titulo}>Calculadora de Notas Medio/TÃ©cnico</Text>

        
        <Text>{resultado}</Text>

        <View style={Styles.ContainerNotas}>
          <Text style={Styles.textoNotas}>N1</Text>
          <TextInput 
            style={Styles.input} 
            placeholder='Digite a primeira nota' 
            value={nota1} 
            onChangeText={setNota1}  
          />

          <Text style={Styles.textoNotas}>N2</Text>
          <TextInput
             style={Styles.input}
             placeholder='Digite a segunda nota'
             value={nota2}
             onChangeText={setNota2}
          />

          <Text style={Styles.textoNotas}>N3</Text>
          <TextInput 
             style={Styles.input}
             placeholder='Digite a terceira nota'
             value={nota3}
             onChangeText={setNota3}
          />

          <Text style={Styles.textoNotas}>N4</Text>
          <TextInput 
             style={Styles.input}
             placeholder='Digite a quarta nota'
             value={nota4}
             onChangeText={setNota4}
          />

          <Text style={Styles.textoNotas}>FINAL</Text>
          <TextInput 
             style={Styles.input}
             value={final}
             onChangeText={setFinal}
          />

          <TouchableHighlight 
            style={Styles.btCalcular}
            onPress={() => Calcular()} testID='btn2'
          >
            <Text style={Styles.textoBt}>Calcular</Text>
          </TouchableHighlight>

          <TouchableHighlight 
            style={Styles.btCalcular}
            onPress={() => Limpar()}
          >
            <Text style={Styles.textoBt}>Limpar</Text>
          </TouchableHighlight>

          <Text>{resultado}</Text>
        
        </View>
        </ScrollView>
    </View>
  )
}