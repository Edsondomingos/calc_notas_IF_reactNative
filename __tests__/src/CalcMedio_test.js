import { render, fireEvent } from "@testing-library/react-native";
import CalcMedioTecnico from '../../components/CalcMedioTecnico'
jest.mock("react-native/Libraries/Animated/NativeAnimatedHelper");

describe("Teste de inicialização e Exibição", ()=>{
    it("Component view CalcMedio", ()=>{
        const { getByText, getByTestId } = render(<CalcMedioTecnico />);

        expect(getByText("Calculadora de Notas Medio/Técnico")).toBeTruthy();
        expect(getByTestId("btn1")).toBeTruthy();
    })
})
describe("Teste de Navegação", ()=>{
    it("Navegação botão calcular", ()=>{
        const { getByText, getByTestId } = render(<CalcMedioTecnico />);

        const botao = getByTestId("btn2");
        fireEvent.press(botao);

        expect(getByText("Calculadora de Notas Medio/Técnico")).toBeTruthy();
    })
})