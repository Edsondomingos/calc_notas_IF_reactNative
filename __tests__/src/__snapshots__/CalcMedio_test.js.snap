// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de Snapshot Snapshot de CalcMedio 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
  testID="teste1"
>
  <RCTScrollView>
    <View>
      <Text
        style={
          Object {
            "fontFamily": "Cursive",
            "fontSize": 30,
            "fontWeight": "bold",
            "marginVertical": 20,
            "textAlign": "center",
          }
        }
      >
        Calculadora de Notas Medio/TÃ©cnico
      </Text>
      <Text />
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "display": "flex",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          N1
        </Text>
        <TextInput
          onChangeText={[Function]}
          placeholder="Digite a primeira nota"
          style={
            Object {
              "borderRadius": 7,
              "borderWidth": 1,
              "margin": "auto",
              "marginBottom": 20,
              "padding": 5,
              "width": 200,
            }
          }
          value=""
        />
        <Text
          style={
            Object {
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          N2
        </Text>
        <TextInput
          onChangeText={[Function]}
          placeholder="Digite a segunda nota"
          style={
            Object {
              "borderRadius": 7,
              "borderWidth": 1,
              "margin": "auto",
              "marginBottom": 20,
              "padding": 5,
              "width": 200,
            }
          }
          value=""
        />
        <Text
          style={
            Object {
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          N3
        </Text>
        <TextInput
          onChangeText={[Function]}
          placeholder="Digite a terceira nota"
          style={
            Object {
              "borderRadius": 7,
              "borderWidth": 1,
              "margin": "auto",
              "marginBottom": 20,
              "padding": 5,
              "width": 200,
            }
          }
          value=""
        />
        <Text
          style={
            Object {
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          N4
        </Text>
        <TextInput
          onChangeText={[Function]}
          placeholder="Digite a quarta nota"
          style={
            Object {
              "borderRadius": 7,
              "borderWidth": 1,
              "margin": "auto",
              "marginBottom": 20,
              "padding": 5,
              "width": 200,
            }
          }
          value=""
        />
        <Text
          style={
            Object {
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          FINAL
        </Text>
        <TextInput
          onChangeText={[Function]}
          style={
            Object {
              "borderRadius": 7,
              "borderWidth": 1,
              "margin": "auto",
              "marginBottom": 20,
              "padding": 5,
              "width": 200,
            }
          }
          value=""
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#21AEEF",
              "borderRadius": 20,
              "height": 45,
              "marginHorizontal": 10,
              "marginVertical": 20,
              "width": 120,
            }
          }
          testID="teste2"
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 25,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Calcular
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#21AEEF",
              "borderRadius": 20,
              "height": 45,
              "marginHorizontal": 10,
              "marginVertical": 20,
              "width": 120,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 25,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Limpar
          </Text>
        </View>
        <Text />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
